tests/test_ad.py

```diff
@@ -210,6 +234,14 @@ def getConfig(self):
 
     @defer.inlineCallbacks
     def test_correct_pwd(self):
+        return_values = [
+            make_awaitable("@mainuser:test"),
+            make_awaitable("@mainuser:test"),
+            make_awaitable("@nonmainuser/subsidiary.example.org:test"),
+            make_awaitable("@mainuser/subsidiary.example.org:test"),
```

Comment:
The other reason I suggest just using `lambda u: u.lower()` above is because when you set up an array of results to return, the test can become trickier to read as you have to keep this in your head.
Sometimes that's justified, but I think the test code would be actually quite happy with just lowercasing the input (maybe I'm wrong though).